src/PostgREST/DbStructure.hs

```diff
@@ -555,7 +557,7 @@ allColumns tabs =
         JOIN pg_enum e ON t.oid = e.enumtypid
         JOIN pg_catalog.pg_namespace n ON n.oid = t.typnamespace
         GROUP BY s,n
-    ) AS enum_info ON (info.udt_name = enum_info.n)
+    ) AS enum_info ON (trim(leading '_' from info.udt_name) = enum_info.n)
```

Comment:
Hmm. Would like to see an example output of this, not sure why the `trim leading` is needed.

Reply:
The `udt_name` (actually `pg_type`.`typname` [pg_type docs](https://www.postgresql.org/docs/current/catalog-pg-type.html)) for enum arrays has a leading `_`, see the [`CREATE TYPE` docs](https://www.postgresql.org/docs/current/sql-createtype.html#id-1.9.3.94.5.9):
> Whenever a user-defined type is created, PostgreSQL automatically creates an associated array type, whose name consists of the element type's name prepended with an underscore, and truncated if necessary to keep it less than `NAMEDATALEN` bytes long. (If the name so generated collides with an existing type name, the process is repeated until a non-colliding name is found.) This implicitly-created array type is variable length and uses the built-in input and output functions `array_in` and `array_out`. The array type tracks any changes in its element type's owner or schema, and is dropped if the element type is.

But your comment about 'Example output' is right in another way at least for the Swagger UI: https://github.com/PostgREST/postgrest/issues/1353#issuecomment-515175906

Reply:
Ok before Postgresql `8.3` this was a valid solution, but my solution is no longer valid, see the [`CREATE TYPE` docs the Notes section](https://www.postgresql.org/docs/current/sql-createtype.html#SQL-CREATETYPE-NOTES):
> Before PostgreSQL version 8.3, the name of a generated array type was always exactly the element type's name with one underscore character (`_`) prepended. (Type names were therefore restricted in length to one less character than other names.) While this is still usually the case, the array type name may vary from this in case of maximum-length names or collisions with user type names that begin with underscore. Writing code that depends on this convention is therefore deprecated. Instead, use `pg_type.typarray` to locate the array type associated with a given type.
>
> It may be advisable to avoid using type and table names that begin with underscore. While the server will change generated array type names to avoid collisions with user-given names, there is still risk of confusion, particularly with old client software that may assume that type names beginning with underscores always represent arrays.